### DISCUSSION TITLE
Token lifetime configuration

### DISCUSSION THREAD

**Discussion by mffap** [zitadel/zitadel#2129] [Ideas]
FAQ: Is there a token lifetime configuration option (auth token, refresh token and id token)?

**Comment by mffap** [comment]
Not yet, but if there's an immediate need we can prioritize this feature. Upvote this idea.

**Comment by sholzmueller** [comment]
Looks like this is implemented by now: https://zitadel.com/docs/guides/manage/console/instance-settings#oidc-token-lifetimes-and-expiration

But it is an instance-wide setting, not configurable per organization. I'd like to see the latter because some organizations are very sensitive about security and ask for short "session timeouts", but I do not want to inconvenience all users with short token lifetimes.

@mffap does it sound like a reasonable feature?

**Reply by fforootd** [reply]
Yes the token lifetimes have been implemented on an instance level. It is totally reasonable to define those values on an org. level as well. We even considered project granularity once 😁 

I just created for https://github.com/zitadel/zitadel/issues/5219 for further tracking of this.

**Reply by sholzmueller** [reply]
Happy to hear that! Granularity per project (or even application) could actually be useful too, but the most obvious override would be on the org level, yes. Thank you for tracking the issue!

**Reply by ckhatton-pvfscaffolding** [reply]
The documentation has changed; the new location is here: https://zitadel.com/docs/guides/manage/console/default-settings#oidc-token-lifetimes-and-expiration

Although, it only just shows a screenshot 😅 Luckily, the actual settings page has instructions on what each setting does.

**Comment by gpoulin** [comment]
Having per client token lifetime can also be useful. One example is for creating long-lived token with limited scope using token-exchange for async processing.

**Reply by fforootd** [reply]
Yeah, we track that with this issue https://github.com/zitadel/zitadel/issues/5219
